Dockerfile

```diff
@@ -42,20 +42,31 @@ ARG TARGETARCH
 
 # Install runtime OS package dependencies
 RUN --mount=type=cache,target=/var/cache/apt \
-    apt-get update && \
+    apt update && \
         # Install dependencies to add additional repos
-        apt-get install -y --no-install-recommends \
+        apt install -y --no-install-recommends \
             # Runtime packages (groff-base is necessary for AWS CLI help)
             ca-certificates curl gnupg git make openssl tar pixz zip unzip groff-base iputils-ping nss-passwords procps iproute2
 
 # FIXME Node 18 actually shouldn't be necessary in Community, but we assume its presence in lots of tests
+# Note: We're extracting and modifying the `control.tar.xz` metadata of the `nodejs` package below,
+#       to remove the dependency to `python3` (to avoid installing python3.9 into the image)
 RUN --mount=type=cache,target=/var/cache/apt \
     mkdir -p /etc/apt/keyrings && \
     curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
     echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_18.x nodistro main" > /etc/apt/sources.list.d/nodesource.list && \
-    apt-get update && \
-    apt-get install -y --no-install-recommends \
-        nodejs
+    apt update && apt install -y binutils xz-utils && \
+    mkdir -p /tmp/nodejs.deb && cd /tmp/nodejs.deb && \
+    apt download nodejs && \
+    ar -xf nodejs*.deb && tar -xf control.tar.xz && \
+    sed -i 's/python3,//g' control && \
+    tar cfJ control.tar.xz control && ar r nodejs*.deb control.tar.xz && \
+    dpkg -i nodejs*.deb && \
+    rm -rf /tmp/*.deb && apt remove --purge -y binutils xz-utils
```

Comment:
This is a pretty sophisticated solution. But maybe it would make sense to directly jump to a installation where we don't automatically install all dependencies? Maybe we could find an archived distribution which we can just extract and link?

Reply:
We could for example just jump to the binary archive installation method: https://github.com/nodejs/help/wiki/Installation
Something like this:
```
# get the latest 18.x version
NODE_ARCH=$([[ "${TARGETARCH}" == "amd64" ]] && echo "x64" || echo "${TARGETARCH}");  \
LATEST_VERSION_FILENAME=`curl -s https://nodejs.org/dist/latest-v18.x/SHASUMS256.txt | grep -o "node-v.*-linux-${NODE_ARCH}"`; \
mkdir -p /usr/local/lib/nodejs; \
curl -s -L https://nodejs.org/dist/latest-v18.x/${LATEST_VERSION_FILENAME}.tar.gz | tar -xz -C /usr/local/lib/nodejs; \
...
```
But I guess this is really a question of taste?

Reply:
Also looks good to me, thanks for digging more into it. Probably more stable to fetch the dist binary directly :+1: